src/pudl/workspace/datastore.py

```diff
@@ -274,9 +283,11 @@ def get_descriptor(self: Self, dataset: str) -> DatapackageDescriptor:
         doi = self.get_doi(dataset)
         if doi not in self._descriptor_cache:
             dpkg = self._fetch_from_url(self._get_url(doi))
-            for f in dpkg.json()["files"]:
-                if f["filename"] == "datapackage.json":
-                    resp = self._fetch_from_url(f["links"]["download"])
+            dpkg.json()["entries"]
+            self._descriptor_cache[doi] = self._descriptor_from
+            for f in dpkg.json()["entries"]:
+                if f["key"] == "datapackage.json":
+                    resp = self._fetch_from_url(f["links"]["content"])
```

Comment:
If our whole setup is contingent on being able to construct a path to a file given the filename and record ID, is there a reason to use the API to obtain the `datapackage.json`?

I guess if we want to verify the checksum on `datapackage.json` we need to get the checksum from Zenodo, since the file can't contain its own checksum.

Reply:
My reason for doing this is that "keeping the datapackage.json around lets me not think too hard about how to construct a `DatapackageDescriptor`."